include/fe/fe_base.h

```diff
@@ -218,9 +226,18 @@ class FEGenericBase : public FEAbstract
   { libmesh_assert(!calculations_started || calculate_dphi);
     calculate_dphi = calculate_dphiref = true; return dphi; }
 
+  const std::vector<std::vector<OutputGradient>> & get_dual_dphi() const
+  { libmesh_assert(!calculations_started || calculate_dphi);
+    calculate_dphi = calculate_dual = calculate_dphiref = true; return dual_dphi; }
+
   void request_dphi() const override
   { get_dphi(); }
 
+  void request_dual_dphi() const override
+  { get_dual_dphi(); }
+
+  const DenseMatrix<Real> get_dual_coeff() const;
```

Comment:
Looks like this isn't defined? I gather this is not a complete PR yet...

Reply:
Thanks! Nice catch! This should be fixed now.